make.js

```diff
@@ -0,0 +1,19 @@
+make = function(...array) {
+    const args = [];
+    const inner = (...temp) => {
+        if(typeof temp[0] === 'function') {
+            return args.reduce(temp[0]);
+        }
+        else {
+            args.push(...temp);
+            return inner;
+        }
+    };
+    return inner(...array);
```

Comment:
When you declare an array in function params, it already became an array, you don't need to destruct it again and again to join it in array in function `inner`